src/mpl2/src/rtl_mp.cpp

```diff
@@ -121,6 +123,95 @@ bool MacroPlacer2::place(const int max_num_macro,
   return true;
 }
 
+odb::dbOrientType MacroPlacer2::stringToOrientType(
+    const std::string& orientation_string)
+{
+  odb::dbOrientType orientation;
+
+  if (orientation_string == "R0") {
+    orientation = odb::dbOrientType::R0;
+  } else if (orientation_string == "MY") {
+    orientation = odb::dbOrientType::MY;
+  } else if (orientation_string == "MX") {
+    orientation = odb::dbOrientType::MX;
+  } else if (orientation_string == "R180") {
+    orientation = odb::dbOrientType::R180;
+  } else {
+    logger_->error(MPL, 33, "Invalid orientation {}.", orientation_string);
+  }
```

Comment:
dbOrientType::dbOrientType(const char* orient);  already does this for you.

Reply:
This function returns only the orientations we allow in mpl2. (we don't allow rotation so, `R90`, `R270` and `MYR90`, `MXR90` are not allowed).

If I used the `dbOrientType::dbOrientType(const char* orient)` I would have to check for the illegal ones either way.

Reply:
While we wouldn't generate a placement with a 90 degree rotation there is no reason a user couldn't chose to that if they knew it was ok.  I think it is at most a warning.

In general I'd prefer to avoid duplication and check the dbOrientType.  You could add a a method there for isRightAngleRotation() that could be general purpose.